### PR TITLE
Vk.com dynamic retargeting pixel GTM Custom Template

### PR DESCRIPTION
**Pull request by d9i2r2t1**
Vk.com is Russian social network with huge marketing opportunities.
Using my GTM Custom Template you can add vk.com dynamic retargeting pixel to your site.
You can use Google Enhanced Ecommerce data via dataLayer to transfer products and event parameters to the pixel.

### CODE REVIEW

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+
+    //get site search phrase method
+    getSiteSearchPhrase: () => {
+        if (settings.event === 'view_search') return getQueryParameters(settings.siteSearchQueryParam);
+        else return undefined;
+    },
+
+    //get event parameters method
+    getEventParams: (products, currencyCode, revenue) => {
+        let eventParamsClean= {};
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `const eventParamsClean = {};`

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+        let eventParams = {
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `const eventParams = {`

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+
+    //get price list ID method
+    getPriceListId: hostname => {
+        if (settings.fewPriceLists) return settings.priceListIds[hostname];
+        else return settings.priceListId;
+    },
+
+    //VK openapi.js init method
+    openapiInit: () => {
+        injectScript('https://vk.com/js/api/openapi.js?159', pixel.setVkAsyncInit(), data.gtmOnFailure, 'vkPixel');
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `injectScript('https://vk.com/js/api/openapi.js?159', pixel.setVkAsyncInit, data.gtmOnFailure, 'vkPixel');`

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 4]
```diff
@@ -0,0 +1,729 @@
...
+
+    //start attempts counter
+    count: 0
+};
+
+//object with methods for processing an array of event products
+const eventProducts = {
+    //get product parameters method
+    getProductParams: products => {
+        let arr = [];
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `const arr = [];`

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 5]
```diff
@@ -0,0 +1,729 @@
...
+        products.forEach(i => {
+            let productParamsClean = {};
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `const productParamsClean = {};`

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 6]
```diff
@@ -0,0 +1,729 @@
...
+            let productParams = {
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `const productParams = {`

File: tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl [hunk 7]
```diff
@@ -0,0 +1,729 @@
...
+            if (productParams.price !== 0) productParamsClean.price = productParams.price;
+            arr.push(productParamsClean);
+        });
+        return arr;
+    },
+
+    //get unique product categories method. Return a string. String type is 'a,b,c'
+    getCategoryString: products => {
+        let categoryId = '';
+        let check = [];
```
Comment (sahava) [inline tags/Vk/DynamicRetargetingPixel/vk_pixel_v1.0_eng.tpl]: `const check = [];`

**Review by sahava — COMMENTED** [review]
Good stuff! Take a look at the couple of style comments (and the fix to the injectScript) and I'll merge once fixed :)

**Comment by d9i2r2t1** [comment]
Thanks for code review! All issues have been fixed. I hope now everything is ok =)

**Pull request merged**